File: JavaScript/프로그래머스/1/12982. 예산/예산.js
function solution(d, budget) {
    
    let sum = 0;
    let answer = 0;
    
    d.sort((a, b) => a - b);
    
    for (let i = 0; i < d.length; i++){
        // 더하기 전에 예산 초과 확인 필요
        if (sum + d[i] <= budget) {
            sum += d[i];
            answer++;
        } else {
            break; 
        }
    }
    
    return answer;
}

File: Java/프로그래머스/1/12982. 예산/예산.java
import java.util.*;

class Solution {
    public int solution(int[] d, int budget) {
        int answer = 0;
        int sum_cost = 0;
        
        Arrays.sort(d);
        
        for (int i = 0; i < d.length; i++) {
            if (sum_cost + d[i] > budget) {
                break;
            }
            sum_cost += d[i];
            answer += 1;
        }
        return answer;
    }
}